src/CustomElements/v1/CustomElements.js

```diff
@@ -0,0 +1,597 @@
+          var namespace = mutation.attributeNamespace;
+          target['attributeChangedCallback'](name, oldValue, newValue, namespace);
+        }
+      }
+    },
+  }
+
+  // Closure Compiler Exports
+  window['CustomElementsRegistry'] = CustomElementsRegistry;
+  CustomElementsRegistry.prototype['define'] = CustomElementsRegistry.prototype.define;
+  CustomElementsRegistry.prototype['get'] = CustomElementsRegistry.prototype.get;
+  CustomElementsRegistry.prototype['whenDefined'] = CustomElementsRegistry.prototype.whenDefined;
+  CustomElementsRegistry.prototype['flush'] = CustomElementsRegistry.prototype.flush;
+  CustomElementsRegistry.prototype['polyfilled'] = CustomElementsRegistry.prototype.polyfilled;
+  CustomElementsRegistry.prototype['enableFlush'] = CustomElementsRegistry.prototype.enableFlush;
+
+  // patch window.HTMLElement
+
+  var origHTMLElement = win.HTMLElement;
+  win.HTMLElement = function HTMLElement() {
+    var customElements = win['customElements'];
+    if (customElements._newInstance) {
+      var i = customElements._newInstance;
+      customElements._newInstance = null;
+      return i;
+    }
+    if (this.constructor) {
+      var tagName = customElements._constructors.get(this.constructor);
+      return doc._createElement(tagName, false);
+    }
+    throw new Error('unknown constructor. Did you call customElements.define()?');
+  }
+  win.HTMLElement.prototype = Object.create(origHTMLElement.prototype);
+  Object.defineProperty(win.HTMLElement.prototype, 'constructor', {value: win.HTMLElement});
+
+  // patch all built-in subclasses of HTMLElement to inherit from the new HTMLElement
+  // See https://html.spec.whatwg.org/multipage/indices.html#element-interfaces
+
+  /** @const */
+  var htmlElementSubclasses = [
+  	'Button',
+  	'Canvas',
+  	'Data',
+  	'Head',
+  	'Mod',
+  	'TableCell',
+  	'TableCol',
+    'Anchor',
+    'Area',
+    'Base',
+    'Body',
+    'BR',
+    'DataList',
+    'Details',
+    'Dialog',
+    'Div',
+    'DList',
+    'Embed',
+    'FieldSet',
+    'Form',
+    'Heading',
+    'HR',
+    'Html',
+    'IFrame',
+    'Image',
+    'Input',
+    'Keygen',
+    'Label',
+    'Legend',
+    'LI',
+    'Link',
+    'Map',
+    'Media',
+    'Menu',
+    'MenuItem',
+    'Meta',
+    'Meter',
+    'Object',
+    'OList',
+    'OptGroup',
+    'Option',
+    'Output',
+    'Paragraph',
+    'Param',
+    'Picture',
+    'Pre',
+    'Progress',
+    'Quote',
+    'Script',
+    'Select',
+    'Slot',
+    'Source',
+    'Span',
+    'Style',
+    'TableCaption',
+    'Table',
+    'TableRow',
+    'TableSection',
+    'Template',
+    'TextArea',
+    'Time',
+    'Title',
+    'Track',
+    'UList',
+    'Unknown',
+  ];
+
+  for (var i = 0; i < htmlElementSubclasses.length; i++) {
+    var ctor = window['HTML' + htmlElementSubclasses[i] + 'Element'];
+    if (ctor) {
+      ctor.prototype.__proto__ = win.HTMLElement.prototype;
+    }
+  }
+
+  // patch doc.createElement
```

Comment:
It's critical to also patch `document.importNode` to ensure synchronous upgrade of stamped templates.